op-service/eth/types.go

```diff
@@ -619,3 +620,42 @@ const (
 	GetPayloadV2 EngineAPIMethod = "engine_getPayloadV2"
 	GetPayloadV3 EngineAPIMethod = "engine_getPayloadV3"
 )
+
+// StorageKey is a marshaling utility for hex-encoded storage keys, which can have leading 0s and are
+// an arbitrary length.
+type StorageKey []byte
+
+func (k *StorageKey) UnmarshalJSON(text []byte) error {
+	// unmarshal string that may have 0x prefix
+	var str string
+	if err := json.Unmarshal(text, &str); err != nil {
+		return err
+	}
+
+	if len(str)%2 != 0 {
+		// add leading 0 if odd length
+		if strings.HasPrefix(str, "0x") {
+			str = str[:2] + "0" + str[2:]
+		} else {
+			str = "0" + str
+		}
+	}
+
+	// decode hex string
+	b, err := hexutil.Decode(str)
+	if err != nil {
+		fmt.Println("here")
```

Comment:
Stray println got left behind.

Reply:
sorry about that - really need to check every time I submit for review. Fixed now...